apps/OpenSpace/ext/launcher/src/sgctedit/monitorbox.cpp

```diff
@@ -0,0 +1,216 @@
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE  *
+ * OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.                                         *
+ ****************************************************************************************/
+
+#include "sgctedit/monitorbox.h"
+
+MonitorBox::MonitorBox(QRect widgetDims, std::vector<QRect> monitorResolution,
+                         unsigned int nWindows, QString* winColors)
+    : _monitorWidgetSize(widgetDims)
+    , _monitorResolution(monitorResolution)
+    , _nWindows(nWindows)
+    , _colorsForWindows(winColors)
+{
+    _nMonitors = monitorResolution.size();
+    _showLabel = (_nMonitors > 1);
+    mapMonitorResolutionToWidgetCoordinates();
+}
+
+MonitorBox::~MonitorBox()
+{
+}
+
+void MonitorBox::paintEvent(QPaintEvent *event)
+{
+    Q_UNUSED(event);
+    QPainter painter(this);
+    QPen pen = painter.pen();
+    painter.setPen(pen);
+    paintWidgetBorder(painter, width(), height());
+    //Draw window out-of-bounds region(s) first
+    for (unsigned int i = 0; i < _nWindows; ++i) {
+        paintWindowBeyondBounds(painter, i);
+    }
+    //Draw & fill monitors over the out-of-bounds regions
+    paintMonitorBackgrounds(painter);
+    //Draw window(s) over both out-of-bounds and monitors
+    for (unsigned int i = 0; i < _nWindows; ++i) {
+        paintWindow(painter, i);
+    }
+}
+
+void MonitorBox::paintWidgetBorder(QPainter& painter, const int width, const int height) {
+    painter.setPen(QPen(Qt::gray, 4));
+    painter.drawRoundedRect(0, 0, width - 1, height - 1, 10, 10);
+}
+
+void MonitorBox::paintMonitorBackgrounds(QPainter& painter) {
+    painter.setPen(QPen(Qt::black, 2));
+    painter.setFont(QFont("Arial", 14));
+    for (unsigned int i = 0; i < _nMonitors; ++i) {
+        if (i <= _monitorDimensionsScaled.size()) {
+            painter.drawRect(_monitorDimensionsScaled[i]);
+            QColor fillColor("#DDDDDD");
+            QBrush brush(fillColor);
+            brush.setStyle(Qt::SolidPattern);
+            painter.fillRect(_monitorDimensionsScaled[i], brush);
+            if (_showLabel) {
+                QPointF textPos = QPointF(_monitorDimensionsScaled[i].left() + 5,
+                    _monitorDimensionsScaled[i].top() + 18);
+                painter.drawText(textPos, QString::fromStdString(std::to_string(i + 1)));
+            }
+        }
+    }
+}
+
+void MonitorBox::paintWindowBeyondBounds(QPainter& painter, const unsigned int winIdx) {
+    painter.setBrush(Qt::BDiagPattern);
+    setPenSpecificToWindow(painter, winIdx, false);
+    if (winIdx <= _windowRendering.size()) {
+        painter.drawRect(_windowRendering[winIdx]);
+    }
+    setPenSpecificToWindow(painter, winIdx, true);
+    painter.setBrush(Qt::NoBrush);
+}
+
+void MonitorBox::paintWindow(QPainter& painter, const unsigned int winIdx) {
+    setPenSpecificToWindow(painter, winIdx, true);
+    if (winIdx <= _windowRendering.size()) {
```

Comment:
`winIdx` should be `size_t` or else it throws a warning here

Reply:
I'll make this change, but I'm not getting a compiler warning from this

Reply:
Visual studio is throwing warnings left and right about these things.  Didn't mention the others since I was planning to go in and fix those warnings myself, as it would be much easier since I'm compiling directly on it